core/src/test/scala/integration/kafka/api/AdminClientTest.scala

```diff
@@ -76,6 +90,15 @@ class AdminClientTest extends IntegrationTestHarness with Logging {
     assertEquals("consumer", group.protocolType)
   }
 
+  @Test(expected = classOf[TimeoutException])
+  def testListGroupsTimeout() {
+    val map = client.listAllGroups()
```

Comment:
This seems indirect. Would it not be possible to use listGroups() directly with a mocked instance of Node?